admin/modules/forum/management.php

```diff
@@ -1989,11 +1989,25 @@
 				$db->delete_query("forums", "CONCAT(',',parentlist,',') LIKE '%,$fid,%'");
 		}
 
-		$db->delete_query("threads", "fid='{$fid}' {$delquery}");
-		$db->delete_query("posts", "fid='{$fid}' {$delquery}");
+		require_once MYBB_ROOT.'inc/class_moderation.php';
+		$moderation = new Moderation();
+
+		$query = $db->simple_select("threads", "tid", "fid='{$fid}' {$delquery}");
+		while($tid = $db->fetch_field($query, 'tid'))
+		{
+			$moderation->delete_thread($tid);
+		}
+
+		$query = $db->simple_select('posts', 'pid', "fid='{$fid}' {$delquery}");
+		while($pid = $db->fetch_field($query, 'pid'))
+		{
+			$moderation->delete_post($pid);
+		}
```

Comment:
Yep, not needed. Except that, you forgot about `forumsread`: https://github.com/mybb/mybb/blob/feature/install/resources/mysql_db_tables.php#L308 And the old code that updates global stats is not needed anymore due to: https://github.com/mybb/mybb/blob/50e5a961708a004457dad05589db2d2770b0419c/inc/functions.php#L2568